core/libp2p/peer/peer_identity.hpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_PEER_IDENTITY_HPP
+#define KAGOME_PEER_IDENTITY_HPP
+
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include <outcome/outcome.hpp>
+#include "libp2p/multi/multiaddress.hpp"
+#include "libp2p/peer/peer_info.hpp"
+
+namespace libp2p::peer {
+  /**
+   * Identity of the given peer; includes its ID and multiaddress; usually is
+   * passed over the network
+   */
+  class PeerIdentity {
+    using FactoryResult = outcome::result<PeerIdentity>;
+
+   public:
+    enum class FactoryError { ID_EXPECTED = 1, NO_ADDRESSES, SHA256_EXPECTED };
+
+    /**
+     * Create a PeerIdentity from the string of format
+     * "<multiaddress>/id/<base58_encoded_peer_id>"
```

Comment:
change /id to /p2p (I asked Raul, he confirmed this token)

Reply:
P.S. you can reuse 
```
#include "libp2p/multi/multiaddress.hpp"

Protocol::get(Protocol::Code::p2p)
```

Reply:
There's no "p2p" protocol in the Multiadress' protocol list :(